### DISCUSSION TITLE
Imaginary frequency is reduced but not disappeared after the step of removing the imaginary frequency

### DISCUSSION THREAD

**Discussion by jgSi1113** [SSCHAcode/python-sscha#349] [General]
Dear all,

I want to check the anharmonic effect and the potential quantum atomic fluctuations of my system, where the harmonic phonon shows some imaginary modes at some non-gamma points.  When I finished the harmonic dynamic matrix calculations and removed the the imaginary frequency by _dyn.Symmetrize()  &   dyn.ForcePositiveDefinite()_, I found the imaginary frequency is reduced, but still existed in a very small zone.  How to deal with it?  Does it mean the system is unstable? 
![image](https://github.com/user-attachments/assets/644c9b75-a82d-4bec-bf56-d50d9b496308)

Best & Regards!
Jianguo

**Comment by mesonepigreco** [comment]
Hi, by using ``dyn.ForcePositiveDefinite()`` you only enforce the q-points in the q-mesh to be positive, and this is enough for the SSCHA (e.g., the R point is commensurate with an even grid, so it becomes positive, but not necessary what is close to R does).

You can perform the SSCHA calculation to see what happens to the final converged result. If it still has imaginary phonons when interpolated on this path, then it is probably a signature that the q-mesh (or supercell) employed is insufficient to converge this system.

**Reply by jgSi1113** [reply]
Thanks for your reply. I will start the SSCHA calculations. Hope I can obtain some good results.

Best & Regards!
Jianguo
